src/main/io/osd.c

```diff
@@ -55,522 +60,290 @@
-    {"PID PROFILE", OME_UINT8, NULL, &entryPidProfile},
-    {"RATE & RXPO", OME_Submenu, osdChangeScreen, &menuRateExpo[0]},
-    {"RC PREVIEW", OME_Submenu, osdChangeScreen, &menuRc[0]},
-    {"MISC", OME_Submenu, osdChangeScreen, &menuMisc[0]},
-    {"BACK", OME_Back, NULL, NULL},
-    {NULL, OME_END, NULL, NULL}
-};
+        case OSD_CRAFT_NAME:
+        {
+            if (strlen(masterConfig.name) == 0)
+                strcpy(buff, "CRAFT_NAME");
+            else {
+                for (uint8_t i = 0; i < MAX_NAME_LENGTH; i++) {
+                    buff[i] = toupper((unsigned char)masterConfig.name[i]);
+                    if (masterConfig.name[i] == 0)
+                        break;
+                }
+            }
 
-uint8_t tempPid[4][3];
+            break;
+        }
 
-static OSD_UINT8_t entryRollP = {&tempPid[PIDROLL][0], 10, 150, 1};
-static OSD_UINT8_t entryRollI = {&tempPid[PIDROLL][1], 1, 150, 1};
-static OSD_UINT8_t entryRollD = {&tempPid[PIDROLL][2], 0, 150, 1};
+        case OSD_THROTTLE_POS:
+        {
+            buff[0] = SYM_THR;
+            buff[1] = SYM_THR1;
+            sprintf(buff + 2, "%d", (constrain(rcData[THROTTLE], PWM_RANGE_MIN, PWM_RANGE_MAX) - PWM_RANGE_MIN) * 100 / (PWM_RANGE_MAX - PWM_RANGE_MIN));
+            break;
+        }
 
-static OSD_UINT8_t entryPitchP = {&tempPid[PIDPITCH][0], 10, 150, 1};
-static OSD_UINT8_t entryPitchI = {&tempPid[PIDPITCH][1], 1, 150, 1};
-static OSD_UINT8_t entryPitchD = {&tempPid[PIDPITCH][2], 0, 150, 1};
+#ifdef VTX
+        case OSD_VTX_CHANNEL:
+        {
+            sprintf(buff, "CH:%d", current_vtx_channel % CHANNELS_PER_BAND + 1);
+            break;
+        }
+#endif // VTX
 
-static OSD_UINT8_t entryYawP = {&tempPid[PIDYAW][0], 10, 150, 1};
-static OSD_UINT8_t entryYawI = {&tempPid[PIDYAW][1], 1, 150, 1};
-static OSD_UINT8_t entryYawD = {&tempPid[PIDYAW][2], 0, 150, 1};
+        case OSD_CROSSHAIRS:
+        {
+            uint8_t *screenBuffer = max7456GetScreenBuffer();
+            uint16_t position = 194;
 
-OSD_Entry menuPid[] =
-{
-    {"------- PID -------", OME_Label, NULL, NULL},
-    {"ROLL P", OME_UINT8, NULL, &entryRollP},
-    {"ROLL I", OME_UINT8, NULL, &entryRollI},
-    {"ROLL D", OME_UINT8, NULL, &entryRollD},
+            if (maxScreenSize == VIDEO_BUFFER_CHARS_PAL)
+                position += 30;
 
-    {"PITCH P", OME_UINT8, NULL, &entryPitchP},
-    {"PITCH I", OME_UINT8, NULL, &entryPitchI},
-    {"PITCH D", OME_UINT8, NULL, &entryPitchD},
+            screenBuffer[position - 1] = (SYM_AH_CENTER_LINE);
+            screenBuffer[position + 1] = (SYM_AH_CENTER_LINE_RIGHT);
+            screenBuffer[position] = (SYM_AH_CENTER);
 
-    {"YAW P", OME_UINT8, NULL, &entryYawP},
-    {"YAW I", OME_UINT8, NULL, &entryYawI},
-    {"YAW D", OME_UINT8, NULL, &entryYawD},
+            return;
+        }
 
-    {"BACK", OME_Back, NULL, NULL},
-    {NULL, OME_END, NULL, NULL}
-};
+        case OSD_ARTIFICIAL_HORIZON:
```

Comment:
`OSD_ARTIFICIAL_HORIZON` needs to be rewritten not to use screenbuffer, but to just write appropriate characters at correct positions.